sound/soc/sof/topology.c

```diff
@@ -2346,11 +2346,13 @@ static int sof_link_load(struct snd_soc_component *scomp, int index,
 	int ret = 0;
 
 	link->platform_name = "sof-audio";
-	link->nonatomic = true;
 
 	/* send BE configurations to DSP */
-	if (!link->no_pcm)
+	if (!link->no_pcm) {
+		/* set FE link in nonatomic environment */
```

Comment:
FE trigger is not executed in an atomic environment.?

Reply:
BE->no_pcm is true and FE->no_pcm is false. So in our case (!link->no_pcm), FE will be true and will be set nonatomic = 1;

Reply:
And in FE trigger, it is not executed in an atomic environment because sof_ipc_tx_message() needs to be in non-atomic environment.

Reply:
@libinyang Pierre's only asking to re-word your comment to 
" FE trigger is not executed in atomic context". not asking for an explanation

Reply:
Yes, my fault. I reply in my second comments. Thanks @ranj063

Reply:
> @libinyang Pierre's only asking to re-word your comment to
> " FE trigger is not executed in atomic context". not asking for an explanation

@ranj063 @plbossart 
I understand your means...
Sorry for my poor understanding...